### ISSUE TITLE
SLF4J documentation not complete

### ISSUE THREAD

**Issue by dnsbtchr** [OpenFeign/feign#1142]
Hi,
I wanted to make my application log using SLF4J. Therefore I set everything up like documented here https://github.com/OpenFeign/feign (SFL4J part) or here https://github.com/OpenFeign/feign/tree/master/slf4j but it didn't work.
Using this tutorial https://www.baeldung.com/intro-to-feign I found out that it is necessary to set the Feign log level as well and that the default one is NONE. Since the log level can actually be controlled using the SLF4J implementation (e.g. Logback or Log4J) I think it would be good to always set it to `FULL` and then adjust it using the SLF4J ways to do that. Otherwise, what happened to me, people will not get any logging even though they have set the correct log level in their SLF4J configuration.
Even though setting the log level is not really coupled to SLF4J I think it would be good to add setting it to the documentation to avoid confusion.
I would expect something like this:
`GitHub github = Feign.builder()
                     .logger(new Slf4jLogger())
                     .logLevel(Level.FULL)
                     .target(GitHub.class, "https://api.github.com");`
I'm curious how you think about it.
Best
Dennis

**Comment by kdavisk6** [comment]
A few thoughts.

Our `Logger` abstraction is meant to control how much information is logged during a managed operation.  There is no connection between the logger instances level and our `Logger.LogLevel`.  This, I suspect, is the source of your confusion.  

If we were to start today, we would not use this abstraction but adopt SLF4J or Commons Logging, replacing the `Logger` abstraction with a `LogConfiguration` object responsible for controlling what is logged.  This is something we will consider for Feign 11.

For now, your builder example is the correct way to get the behavior you are looking for.  In the meantime, we are open to any suggestions on what we could change `Logger.LogLevel` to in order to indicate what it's actually doing and reduce confusion.

**Comment by dnsbtchr** [comment]
Hi @kdavisk6, 
thanks for the quick response. I totally agree with what you said about the logger abstraction. My only intention when creating this issue was to recommend to include `.logLevel(Level.FULL)` in the 2 examples in the documentation so that in the end it looks like the snipped I provided. 
I think it will help other people in the future to not struggle as much as I did. I can also create a PR for that if you want but since it's really a tiny change in 2 files I thought it might be simpler to just create a ticket.
Best
Dennis
